awsx/utils.ts

```diff
@@ -82,15 +82,12 @@ export function getRegionFromOpts(opts: pulumi.CustomResourceOptions): pulumi.Ou
 
 /** @internal */
 export function getRegion(res: pulumi.Resource): pulumi.Output<aws.Region> {
-  // A little strange, but all we're doing is passing a fake type-token simply to get
-  // the AWS provider from this resource.
-  const provider = res.getProvider ? res.getProvider("aws::") : undefined;
-  return getRegionFromProvider(provider);
+  // uses the provider from the parent resource to fetch the region
+  return aws.getRegionOutput({}, { parent: res }).apply(region => region.name as aws.Region);
```

Comment:
To confirm, as this inheritance is a little tricky, if `res` uses a custom version of the AWS provider, `{ parent: res}` will inherit that custom version so we get the right thing here?

Reply:
Yes, that's my understanding. It actually retrieves the provider configured for the parent. If that one's using a custom version then this will also use that version.

Reply:
I see this is already shipped in aws-apigateway so that's nice! THank you.